File: models/asset.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

const Asset = sequelize.define('Asset', {
  id: {
    type: DataTypes.INTEGER,
    autoIncrement: true,
    primaryKey: true
  },
  asset_tag: {
    type: DataTypes.STRING,
    allowNull: false,
    unique: true
  },
  name: {
    type: DataTypes.STRING,
    allowNull: false
  },
  category_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'asset_categories',
      key: 'id'
    }
  },
  department_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'departments',
      key: 'id'
    }
  },
  section_id: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: 'sections',
      key: 'id'
    }
  },
  status: {
    type: DataTypes.ENUM('Active', 'Under Maintenance', 'Faulty', 'Retired'),
    allowNull: false,
    defaultValue: 'Active'
  },
  purchase_date: {
    type: DataTypes.DATEONLY,
    allowNull: true
  },
  cost: {
    type: DataTypes.DECIMAL(15, 2),
    allowNull: true
  },
  vendor: {
    type: DataTypes.STRING,
    allowNull: true
  },
  warranty_expiry: {
    type: DataTypes.DATEONLY,
    allowNull: true
  },
  created_at: {
    type: DataTypes.DATE,
    allowNull: false,
    defaultValue: DataTypes.NOW
  }
}, {
  tableName: 'assets',
  timestamps: false
});

module.exports = Asset;

File: backend-fastapi/app/models/asset.py
import enum
from sqlalchemy import Column, Integer, String, ForeignKey, Numeric, Date, DateTime, Enum as SQLEnum
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.database import Base


class AssetStatus(str, enum.Enum):
    ACTIVE = "Active"
    UNDER_MAINTENANCE = "Under Maintenance"
    FAULTY = "Faulty"
    RETIRED = "Retired"


class Asset(Base):
    __tablename__ = "assets"

    id = Column(Integer, primary_key=True, index=True)
    asset_tag = Column(String, unique=True, nullable=False, index=True)
    name = Column(String, nullable=False)
    category_id = Column(Integer, ForeignKey("asset_categories.id"), nullable=False)
    department_id = Column(Integer, ForeignKey("departments.id"), nullable=False)
    section_id = Column(Integer, ForeignKey("sections.id"), nullable=False)
    status = Column(SQLEnum(AssetStatus), nullable=False, default=AssetStatus.ACTIVE)
    purchase_date = Column(Date, nullable=True)
    cost = Column(Numeric(15, 2), nullable=True)
    vendor = Column(String, nullable=True)
    warranty_expiry = Column(Date, nullable=True)
    created_at = Column(DateTime(timezone=True), server_default=func.now())

    category = relationship("AssetCategory", backref="assets")
    department = relationship("Department", backref="assets")
    section = relationship("Section", backref="assets")
    maintenance_logs = relationship("MaintenanceLog", back_populates="asset", cascade="all, delete-orphan", passive_deletes=True)
    fault_reports = relationship("FaultReport", back_populates="asset", cascade="all, delete-orphan", passive_deletes=True)